### ISSUE TITLE
AddReportEntry does not support strings the %s in them that are not used for substitution 

### ISSUE THREAD

**Issue by dschveninger** [onsi/ginkgo#1195]
We are using a string to spawn a subprocess cmd that has a %s in it and at the same time trying to report the string using AddReportEntry functions.  
The function is reporting "@ %!s(MISSING)" at the end of the AddReportEntry output and dropping the %.

Would it be possible to use this function and stop any substitution? 

https://github.com/onsi/ginkgo/blob/master/types/report_entry.go#L27

**Comment by onsi** [comment]
hey - sorry, could you share a quick reproducer?  I just tried this:

```
var _ = It("works?", func() {
	AddReportEntry("My %s Report Entry", "some command - %s")
})
```

and confirmed that both the entry name and entry valued were emitted without any `(MISSING)` format strings.

**Comment by dschveninger** [comment]
Here is code example

```
var _ = Describe("Expects examples", Label("issues"), func() {
	Context("Successful test", func() {
		It("message substitution example", func() {
			AddReportEntry("Sample to show what happens with string sub and no arguments", ReportEntryVisibilityFailureOrVerbose)
			cmdToRun := "env envsubst \"$(printf '${%s} ' $(env | sed 's/=.*//'))\" <\".../collector-data-sub.sh\""
			fmt.Println(cmdToRun)
			AddReportEntry(cmdToRun, ReportEntryVisibilityFailureOrVerbose)
		})
	})
})
```
and here is the output, the first bold line if PrintLn and the second one if the AddReportEntry.  We are running 2.6.1 and I will try it after an upgrade.

Expects examples [issues]
/home/dschveninger/nc-api-testing/tests/prototype/current-direction/examples/ginkgo_issues_test.go:11
  Successful test
  /home/dschveninger/nc-api-testing/tests/prototype/current-direction/examples/ginkgo_issues_test.go:12
    message substitution example
    /home/dschveninger/nc-api-testing/tests/prototype/current-direction/examples/ginkgo_issues_test.go:13
  > Enter [It] message substitution example - /home/dschveninger/nc-api-testing/tests/prototype/current-direction/examples/ginkgo_issues_test.go:13 @ 05/12/23 16:22:57.899
  Sample to show what happens with string sub and no arguments - /home/dschveninger/nc-api-testing/tests/prototype/current-direction/examples/ginkgo_issues_test.go:14 @ 05/12/23 16:22:57.899
**env envsubst "$(printf '${%s} ' $(env | sed 's/=.*//'))" <".../collector-data-sub.sh"**
  **env envsubst "$(printf '${/home/dschveninger/nc-api-testing/tests/prototype/current-direction/examples/ginkgo_issues_test.go:17} ' $(env | sed 's/=.*//'))" <".../collector-data-sub.sh" - 05/12/23 16:22:57.9 @** %!s(MISSING)
  < Exit [It] message substitution example - /home/dschveninger/nc-api-testing/tests/prototype/current-direction/examples/ginkgo_issues_test.go:13 @ 05/12/23 16:22:57.9 (0s)
• [0.000 seconds]
------------------------------

**Comment by onsi** [comment]
> We are running 2.6.1 and I will try it after an upgrade.

Ah - that might be it.  Sorry I didn't think to ask about the version earlier.  This was, in fact, fixed in 2.8.1:

https://github.com/onsi/ginkgo/releases/tag/v2.8.1

**Comment by dschveninger** [comment]
Thank you for your quick responses. We will update our version.
